packages/web/src/components/gcds-link/gcds-link.css

```diff
@@ -1,49 +1,72 @@
-:host a {
-  cursor: pointer;
-  color: var(--gcds-link-default);
-  text-decoration: underline;
-  text-underline-offset: var(--gcds-link-underline-offset);
-  text-decoration-color: currentColor;
-  text-decoration-thickness: var(--gcds-link-decoration-thickness);
-
-  &:hover {
-    color: var(--gcds-link-hover);
-    text-decoration-thickness: var(--gcds-link-hover-decoration-thickness);
-  }
+@layer reset, default, display, size, variant, hover, focus;
 
-  &:focus {
-    border-radius: var(--gcds-link-focus-border-radius);
-    background-color: var(--gcds-link-focus-background);
-    color: var(--gcds-link-focus-text);
-    box-shadow: var(--gcds-link-focus-box-shadow);
-    outline: var(--gcds-link-focus-outline-width) solid
-      var(--gcds-link-focus-background);
-    outline-offset: var(--gcds-link-focus-outline-offset);
-    border-color: var(--gcds-link-focus-background);
-    text-decoration: none;
+@layer reset {
+  :host {
+    display: inline-block;
+
+    slot {
+      display: initial;
+    }
   }
+}
 
-  &.link--small {
-    font: var(--gcds-link-font-small);
+@layer default {
+  :host .gcds-link {
+    cursor: pointer;
+    color: var(--gcds-link-default);
+    text-decoration: underline currentColor
+      var(--gcds-link-decoration-thickness);
+    text-underline-offset: var(--gcds-link-underline-offset);
+    transition: all 0.35s;
   }
+}
 
-  &.link--regular {
-    font: var(--gcds-link-font-regular);
+@layer display {
+  :host .gcds-link.d-block {
+    display: block;
   }
+}
 
-  &.link--inherit {
-    font: inherit;
+@layer size {
+  :host .gcds-link {
+    &.link--small {
+      font: var(--gcds-link-font-small);
+    }
+
+    &.link--regular {
+      font: var(--gcds-link-font-regular);
+    }
+
+    &.link--inherit {
+      font: inherit;
+    }
   }
+}
 
-  &.variant-light {
+@layer variant {
+  :host .gcds-link.variant-light {
     color: var(--gcds-link-light);
   }
+}
 
-  &.d-block {
-    display: block;
+@layer hover {
+  @media (hover: hover) {
+    :host .gcds-link:hover {
+      color: var(--gcds-link-hover);
```

Comment:
When `gcds-link` has `variant="light"` the hover state still changes the text colour to dark blue making the text unreadable.

Reply:
Good catch, thank you!